Description: Fix compilation of slusb with 2.6.37
 *_MUTEX got deprecated since kernel 2.6.37
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@sabily.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=613906
Bug-Ubuntu: https://bugs.launchpad.net/bugs/737650
--- a/drivers/st7554.c
+++ b/drivers/st7554.c
@@ -233,7 +233,11 @@
 static struct class *st7554_class;
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,37)
 static DECLARE_MUTEX(open_sem);
+#else
+static DEFINE_SEMAPHORE(open_sem);
+#endif
 
 /* --------------------------------------------------------------------- */
 
@@ -1234,7 +1238,11 @@
 	memset(s, 0, sizeof(*s));
 
 	spin_lock_init(&s->lock);
+#ifdef init_MUTEX
 	init_MUTEX (&s->sem);
+#else
+  sema_init (&s->sem, 1);
+#endif
 	init_waitqueue_head(&s->wait);
 
 	s->name = "ST7554 USB Modem";
